.github/workflows/release.yml

```diff
@@ -0,0 +1,123 @@
+name: Publish Release
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+  workflow_dispatch: {}
+
+jobs:
+  release:
+    name: Release
+    runs-on: ubuntu-latest
+
+    steps:
+      # ToDo when raising PR in Public shared actions
+      # Either a GitHub application or a Bot PAT should be used below
+      # Recommended approach is to use a GitHub App https://docs.github.com/en/apps/oauth-apps/building-oauth-apps/differences-between-github-apps-and-oauth-apps#machine-vs-bot-accounts
+      # Create a GH org Application, create a Private key
+      # Add the private key and app_id as PSA repo secret and use below
+      - name: Generate privileged push token to bypass branch protection settings
+        id: github-app-token
+        uses: tibdex/github-app-token@b62528385c34dbc9f38e5f4225ac829252d1ea92 # v1.8.0
+        with:
+          app_id: ${{ secrets.APP_ID }}
+          private_key: ${{ secrets.PRIVATE_KEY }}
+
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+          token: ${{ steps.github-app-token.outputs.token }}
+
+      - name: Set up Node.js
```

Comment:
To automatically restore global cache in the `setup-dependencies` job, Refer pnpm global package caching at https://github.com/actions/setup-node/blob/main/docs/advanced-usage.md#caching-packages-data

```
- uses: pnpm/action-setup@v2
  with:
    version: <REPLACE>
- uses: actions/setup-node@v4
  with:
    node-version: '<REPLACE>'
    cache: 'pnpm' # automatically restores pnpm global cache but not node_modules from the previous job if available
```